app/models/dialog_field_importer.rb

```diff
@@ -32,12 +32,15 @@ def import_field(dialog_field_attributes)
 
   def set_category_for_tag_control(dialog_field, dialog_field_attributes)
     category_name = dialog_field_attributes["options"][:category_name]
-    if category_name
+    category_id = dialog_field_attributes["options"][:category_id]
+    if category_id
```

Comment:
I can see how this would work in an edit scenario where you expect the tag category to already exist but doesn't this method also get used when importing dialogs between completely different environments?

The category ID may exist in the new environment but it could point to a completely different tag category.  Also, this is just checking that the ID exists in the field data, but there is no check that the category exists when importing into a different environment.

cc @eclarizio

Reply:
@gmcculloug You're right

@syncrou What I think you will need to do is if the category id exists, look up the category and then compare the name/description. The only problem is that in the edit scenario, the name/description that is getting passed in is incorrect as we determined when we were debugging and was the whole source of this issue. I think the best solution in this case would be to modify the ui-components here https://github.com/ManageIQ/ui-components/blob/master/src/dialog-editor/components/modal-field-template/tag-control.html#L26 and add an `ng-change` method that ends up setting the id, name, and description.

Reply:
@eclarizio - Thanks! - I'll get those changes in.